proposals/testing/XXXX-issue-severity-warning.md

```diff
@@ -0,0 +1,113 @@
+# Test Issue Severity
+
+* Proposal: [ST-XXXX](XXXX-issue-severity-warning.md)
+* Authors: [Suzy Ratcliff](https://github.com/suzannaratcliff)
+* Review Manager: TBD
+* Status: **Pitched**
+* Bug: [swiftlang/swift-testing#1075](https://github.com/swiftlang/swift-testing/pull/1075)
+* Implementation: [swiftlang/swift-testing#1075](https://github.com/swiftlang/swift-testing/pull/1075)
+
+## Introduction
+
+I propose introducing a new API to Swift Testing that allows developers to record issues with a specified severity level. By default, all issues will have severity level “error”, and a new “warning” level will be added to represent less severe issues. The effects of the warning recorded on a test will not cause a failure but will be included in the test results for inspection after the run is complete.
+
+## Motivation
+
+Currently, when an issue arises during a test, the only possible outcome is to mark the test as failed. This presents a challenge for users who want a deeper insight into the events occurring within their tests. By introducing a dedicated mechanism to record issues that do not cause test failure, users can more effectively inspect and diagnose problems at runtime and review results afterward. This enhancement provides greater flexibility and clarity in test reporting, ultimately improving the debugging and analysis process.
+
+## Proposed solution
+We propose introducing a new property on `Issue` in Swift Testing called `Severity`, that represents if an issue is a `warning` or an `error`.
+The default Issue severity will still be `error` and users can set the severity when they record an issue.
+
+Test authors will be able to inspect if the issue is a failing issue and will be able to check the severity.
+
+## Detailed design
+
+*Severity Enum*
+
+We introduce a Severity enum to categorize issues detected during testing. This enum is crucial for distinguishing between different levels of test issues and is defined as follows:
+
+The `Severity` enum:
+
+```swift
+  public enum Severity: Sendable {
```

Comment:
Should the other conformances be included? Also, it wasn't obvious that `Issue.Severity` is a nested type.

```suggestion
extension Issue {
  public enum Severity: Codable, Comparable, CustomStringConvertible, Sendable {
```